### ISSUE TITLE
Update version of Activiti

### ISSUE THREAD

**Issue by openwms** [openwms/org.openwms.tms.routing#21]

**Comment by openwms** [comment]
After some research and try out it is the following situation:
- Currently we're using 5.21.0 as WFE as well as for the modeller. This version only supports JUEL1.0 expressions which restricts us in our design. **We must upgrade at least tu EL2.2 in order to use maps, lists and varargs in expressions.**
- The latest release in the 5.x train is 5.22.0. We'll upgrade - but this does not solve the JUEL requirements
- The current latest available GA of Activiti is version 6.x. Unfortunately this comes without any modeller tool. But the modeller is essential for us. At best as web application, directly connected to the database where hot deployments are possible
- The upcoming 7.x release focuses more on cloud deployment that is not really an option for OpenWMS.org. But that version seems to provide a modeller again, called [activiti-modeling-app](https://github.com/Activiti/activiti-modeling-app)

**To sum up:**
Currently no solution to support JUEL2.2 in production.
- Release of 7.x already delayed
- New Modelling App requires another IdP process, keycloak Auth0 whatever, that makes it complicated
- The Activiti integration is wrapped very well, that makes it easy to swap out Activiti by something else, like Camunda. The first research and a quick chat with @berndruecker (Bernd Rücker) at Devoxx2018 pointed out that there is even not a proper replacement for the Activiti 5.x Web Modeller - Just an offline desktop modeller!

//cc @Salaboy @berndruecker @LukasPr

**Comment by salaboy** [comment]
@openwms activiti 6 does come with a modeler tool, where did you got that it doesn’t? 

Also activiti cloud is not really delayed, we are hardening it for a GA release, you can also use activiti core, which is built based on activiti 6 GA. As a community project we usually recommend to adopt beta releases if you are planning to upgrade, due the fact that if you find an issue, we can fix it before GA. 

Other alternatives that should should look at are jBPM and Flowable which are also in the open source space.

**Comment by berndruecker** [comment]
Hi @openwms - that must have been a misunderstanding: There is the Camunda Modeler (https://camunda.com/products/modeler/) as desktop tool. But this is basically an electron wrapping https://bpmn.io/ - which is completely JS based and works in any browser (this is also used upstream to the Activiti Modeler: https://github.com/Activiti/activiti-modeling-app/blob/develop/package.json#L111).

**Comment by salaboy** [comment]
@berndruecker 👍 yep https://bpmn.io/  rocks! I would discourage everyone who tries to reimplement all that functionality.

**Comment by berndruecker** [comment]
Don't get me started Paul ;-)

**Comment by salaboy** [comment]
Thanks god nobody mention OSworkflow (respect) ;)

**Comment by openwms** [comment]
@salaboy Thanks for responding. Simple answer: I couldn't find the successor of the Activiti Explorer web application in version 6. I searched through the repositories and through branches. I am only interested in the modelling part not task or user administration. What do you recommend?

**Comment by salaboy** [comment]
@openwms in version 6 you have a single application with all the functionality: https://github.com/Activiti/Activiti/tree/6.x/modules/activiti-ui

In general if you are only looking to create BPMN models I would recommend something more like the one that we are creating for Activiti Cloud which is just modeling based on BPMN.io in the same line you have the camunda modeller as @berndruecker mention. It really depends on what are you planning to do with it, just use it out of the box, extend it, embed it, integrate it with something else.. 

PS: sorry for the delayed response in the Gitter channel the other day.. we were in the company xmas party that day but we are really open to answer questions and assist everyone who is interested in the projects.

**Comment by openwms** [comment]
Many many thanks to all your input. And no worries about delayed answers, we're doing OSS this may take a while to respond :-) Your input was very helpful and I had a look at various proposed solutions. In a first iteration I decided to go with the Flowable WFE and Modeler, because Flowable seems super aligned with the current code base. Only imports and process definition files needed to be upgraded to fulfill the current requirements.
But anyway. In a next iteration other aspects, than technical JUEL support, is relevant and we may decide to go with an other solution.
